### PR TITLE
add skipUrlPattern option

### PR DESCRIPTION
**Pull request by zivl**
I think this option is in the need of this kind of tool!
Useful for use cases where you need to skip some URLs

### CODE REVIEW

**Comment by zivl** [comment]
@davidtheclark I'll be happy if you could take a look and merge

File: README.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ All options are optional. The options object may contain any of the following pr
 - **skipLocalhost** `{boolean}` - Default: `false`.
   By default, `localhost` links are treated the same as other links, so if your project is not running locally you'll receive a warning.
   If you want to ignore `localhost` links (e.g. `http://localhost/*`, `http://127.0.0.1/*`), set this option to `true`.
+- **skipUrlPatterns** `{Array}` - Array of `String` | `RegExp`.
+  Useful for use cases where you need to skip some URLs (e.g. `[/^http:\/\/(.*)url-to-ignore\.com/]`)
```
Comment (davidtheclark) [inline README.md]:
```suggestion
 A list of patterns for URLs that should be skipped. Each URL will be tested against each pattern, and will be ignored if `new RegExp(pattern).test(url) === true`. For example, with `skipUrls: [/^http:\/\/(.*)url-to-ignore\.com/, 'https://never-check.com']`, links with the URLs `http://www.url-to-ignore.com/foo` and `https://never-check.com/foo/bar` will not be checked.

**Review by davidtheclark — COMMENTED** [review]
Good idea 👍

**Comment by Mythical-ZA** [comment]
Apologies if this is a stupid question, but can skipUrlPattern be used from the command line? 

I'm getting a lot of broken links that are not really broken and would like to exclude them (I don't know why they're being listed as broken):

- warning  Link to https://developer.android.com/training/location/permissions#request-location-access-runtime is dead  no-dead-urls  remark-lint
- warning  Link to https://developer.android.com/jetpack/androidx/releases/biometric is dead                            no-dead-urls  remark-lint
- warning  Link to http://developer.android.com/reference/android/app/Activity.html#ActivityLifecycle is dead           no-dead-urls  remark-lint
- warning  Link to https://developer.android.com/training/location/request-updates is dead                              no-dead-urls  remark-lint

When I navigate to the URLs manually, they load fine. If it is supported, how do I modify my command to include it? I'm running:
`remark -u remark-lint-no-dead-urls -q docs`

Any help would be much appreciated.

**Comment by wooorm** [comment]
It might be more manageable in a configuration file? https://github.com/unifiedjs/unified-engine/blob/main/doc/configure.md

**Pull request merged**